apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/IconCatalog/IconCatalog.tsx

```diff
@@ -62,11 +62,15 @@ const IconCatalogInner: React.FC = () => {
     [searchTerm, searchV0],
   );
 
-  const onInputChange: InputProps['onChange'] = React.useCallback(
-    (e, { value }) => updateSearchDebounced(value),
+  const onInputChange: NonNullable<InputProps['onChange']> = React.useCallback(
```

Comment:
isn't there some better way ? this type gymnastic is kinda complex for the purpose of the code construct.

doesn't have useCallback a generic we could instantiate ?

do I understood it correctl that `InputProps['onChange']` can be undefined and react 18 types useCallback definition cannot be assigned to undefined ?

Reply:
> do I understood it correctly that InputProps['onChange'] can be undefined and react 18 types useCallback definition cannot be assigned to undefined ?

yes, it's something that will have error even with current React types, no matter if it's typed like this or via useCallback generic.

Reply:
> do I understood it correctly that InputProps['onChange'] can be undefined and react 18 types useCallback definition cannot be assigned to undefined ?

yes, it will have error even with current React types, no matter if it's typed like this or via useCallback generic. In some cases I can type each param individually, but not always types needed available from suite

Reply:
> yes, it will have error even with current React types,

why isn't ci failing then ?